lodash.src.js

```diff
@@ -1353,6 +1353,19 @@
       return this;
     }
 
+    /**
+     * Removes every item from the cache.
+     *
+     * @private
+     * @name clear
+     * @memberOf _.memoize.Cache
+     * @returns {Object} Returns the cache object.
+     */
+    function mapClear() {
+      this.__data__ = {};
```

Comment:
Even though `= {}` is simpler it's severing the cache reference. Instead you should iterate over the properties and `delete` (I know gross but if someone wants to avoid that they can use `Map` or replace the implementation to their taste).